projects/storefrontlib/cms-components/navigation/scroll-to-top/scroll-to-top.component.ts

```diff
@@ -53,25 +57,65 @@ export class ScrollToTopComponent implements OnInit {
     this.setConfig();
   }
 
-  protected setConfig(): void {
-    this.componentData.data$.pipe(take(1)).subscribe((data) => {
-      this.scrollBehavior = data.scrollBehavior ?? this.scrollBehavior;
-      this.displayThreshold = data.displayThreshold ?? this.displayThreshold;
-    });
+  @HostListener('window:scroll', ['$event'])
+  onScroll(): void {
```

Comment:
We unfortunately have the tricky situation where we need to maintain the old behaviour and allow "opt-in" for clients to new features using the feature flag mechanism. In this situation, we will need to use the new behaviour only when the feature is enabled (eg. `if(this.featureConfigService.isEnabled('myFeatureOrFix') { ...new behaviour... } else { ...old behaviour ...}`). 

This also is relevant when we add new behaviour to an existing function. We are never sure how users may have extended our library apis so we need to be cautious about changing anything. Feel free to ask me or @Pio-Bar if you are unsure about anything :)

[Here is an example of using these feature flags.](https://github.com/SAP/spartacus/pull/18493/files)

Reply:
New feature flags have been added ;)